### PR TITLE
Pass tests of FindMeetingQueryTests.java

### PR DESCRIPTION
**Pull request by alexanderlopez**

### CODE REVIEW

File: src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -15,9 +15,82 @@
...
+        returnList.add(TimeRange.WHOLE_DAY);
+
+        trimList(returnList, request.getAttendees(), events);
+
+        removeSmallSlots(returnList, request.getDuration());
+
+        return returnList;
+    }
+
+    private void removeSmallSlots(ArrayList<TimeRange> timeList, long duration) {
```
Comment (salehhamadeh) [inline src/main/java/com/google/sps/FindMeetingQuery.java]: "Parameters are most naturally interpreted as inputs to a function. As a result, output parameters are unexpected, which makes code harder to understand at first glance."

timeList is used as an output parameter here. A better approach would be to not mutate the parameter but to create a new list and return it, similar to Java's String.trim() method. If you do that, you wouldn't have to call clone, which would actually make your method more performant.

So you would call this method as follows:
`returnList = removeSmallSlots(returnList, request.getDuration());`

File: src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -15,9 +15,82 @@
...
+        for (TimeRange timeSlot : (ArrayList<TimeRange>) timeList.clone()) {
+            if (timeSlot.duration() < duration) {
+                timeList.remove(timeSlot);
+            }
+        }
+    }
+
+    private void trimList(ArrayList<TimeRange> timeList, Collection<String> participants,
```
Comment (salehhamadeh) [inline src/main/java/com/google/sps/FindMeetingQuery.java]: Same comment here. Prefer immutable parameters.

**Review by salehhamadeh — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -15,9 +15,73 @@
...
+    ArrayList<TimeRange> queryResult;
+
+    public Collection<TimeRange> query(Collection<Event> events,
+            MeetingRequest request) {
+        queryResult = new ArrayList<TimeRange>();
+
+        TimeRange currentRange = TimeRange.WHOLE_DAY;
+
+        PriorityQueue<Event> eventQueue =
+                new PriorityQueue<Event>(new Comparator<Event>() {
```
Comment (Felomeng) [inline src/main/java/com/google/sps/FindMeetingQuery.java:33]: Could use lambda:
https://leetcode.com/problems/top-k-frequent-elements/discuss/156845/java-priority-queue-lambda-expression

File: src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -15,9 +15,73 @@
...
+        });
+
+        for (Event event : events) {
+            if (eventIntersectsGuestlist(event, request.getAttendees())) {
+                eventQueue.add(event);
+            }
+        }
+
+        while (!eventQueue.isEmpty()) {
+            TimeRange currentEvent = eventQueue.poll().getWhen();
```
Comment (Felomeng) [inline src/main/java/com/google/sps/FindMeetingQuery.java:48]: rename currentEvent to something like currentEventTimeRange as it is not an event

**Review by Felomeng — APPROVED** [review]

**Review by salehhamadeh — APPROVED** [review]
Looks good!

**Pull request merged**